include/oneapi/dpl/pstl/utils.h

```diff
@@ -150,6 +150,21 @@ struct __predicate
 template <typename _Comp, typename _Proj>
 using __compare = __predicate<_Comp, _Proj>;
 
+template <typename _F, typename _Proj>
+struct __unary_op
+{
+    //'mutable' is to relax the requirements for a user functor or/and projection type operator() may be non-const
+    mutable _F __f;
+    mutable _Proj __proj;
+
+    template <typename _TValue>
+    decltype(auto)
```

Comment:
Could you explain here why we should have here exactly `decltype(auto)` instead of `auto` in previous implementation?
Which real gaps we fixing in this change?